ai_genomics/analysis/tag_evaluation/make_tag_validation_dataset.py

```diff
@@ -0,0 +1,130 @@
+"""Script to 1) extract openalex concepts from openalex sample and
+2) generate validation dataset to compare tags across
+sample of sample of texts with DBpedia and OpenAlex tags."""
+
+from ai_genomics.getters.data_getters import load_s3_data, get_s3_dir_files, save_to_s3
+
+from ai_genomics import bucket_name, logger
+from typing import List, Union
+import pandas as pd
+import requests
+
+SAMPLES_DIR = get_s3_dir_files(bucket_name, "inputs/ai_genomics_samples/")
+OA_SAMPLE = load_s3_data(
+    bucket_name, "inputs/ai_genomics_samples/ai_genomics_openalex_samples.csv"
+)
+
+
+def make_openalex_api_url(work_url: str) -> str:
+    """Convert openalex sample work id into json-compatible url."""
+    return f"https://api.openalex.org/works/{work_url.split('/')[-1]}"
+
+
+def get_openalex_sample_concepts(oa_sample: pd.DataFrame = OA_SAMPLE) -> pd.DataFrame:
+    """Calls OpenAlex API to get concepts per OpenAlex sample work id.
+        Returns OpenAlex sample with associated concepts, concept
+        scores and concept levels.
+
+    Args:
+        oa_sample: DataFrame of OpenAlex text samples
+    """
+    oa_sample["api_url"] = oa_sample["work_id"].apply(make_openalex_api_url)
+
+    oa_tags = []
+    for sample in list(oa_sample["api_url"]):
+        oa_response = requests.get(sample)
+        if oa_response.ok:
+            if oa_response.headers["Content-Type"] == "application/json":
+                oa_data = oa_response.json()
+                if "concepts" in oa_data.keys():
+                    tags = [oa["display_name"] for oa in oa_data["concepts"]]
+                    scores = [oa["score"] for oa in oa_data["concepts"]]
+                    levels = [oa["level"] for oa in oa_data["concepts"]]
+                    oa_tags.append([sample, tags, scores, levels])
+                else:
+                    logger.info(f"no concepts found for {sample}")
+            else:
+                logger.info("response is not in json")
+        else:
+            logger.info(f"response code not 200 - {oa_response}")
```

Comment:
I'd be tempted to wrap this up into its own function to query for concepts for a single work

Reply:
Two other thoughts:
1. You could unnest this by testing for negative conditions (e.g. `if not oa_response.ok) and then raising the error message immediately.
2. If you hit an error or get a bad response, how do you want that to be represented in the data? Other than specifying here, you could handle it by explicitly setting the `how` parameter in the merge below.